src/common/models/time-shift/time-shift.mocha.ts

```diff
@@ -40,4 +44,89 @@ describe("isValidTimeShift", () => {
     expect(isValidTimeShift("P2M"), "two months").to.be.true;
     expect(isValidTimeShift("P5Y"), "five years").to.be.true;
   });
+
+  describe("constrainToFilter", () => {
+
+    const shift = TimeShift.fromJS("P3D");
+
+    describe("Fixed time filter", () => {
+      it("does not touch if shifted period do not overlap with original", () => {
+        const filter = new FixedTimeFilterClause({
+          reference: "time",
+          values: List.of(new DateRange({ start: new Date("2010-01-01"), end: new Date("2010-01-02") }))
+        });
+        expect(shift.constrainToFilter(filter, Timezone.UTC).equals(shift)).to.be.true;
+      });
+
+      it("returns empty time shift if shifted period overlap with original", () => {
+        const filter = new FixedTimeFilterClause({
+          reference: "time",
+          values: List.of(new DateRange({ start: new Date("2010-01-01"), end: new Date("2010-01-04") }))
+        });
+        expect(shift.constrainToFilter(filter, Timezone.UTC).equals(TimeShift.empty())).to.be.true;
```

Comment:
Maybe name and reuse the result of `TimeShift.empty()`? I'd be quite a nice parity with `shift`.